Make K8s support patch-specific (#1550)
astronomer/docs: software/version-compatibility-reference.md

| Astronomer Platform | Kubernetes             | Astro CLI     | Postgres | Python                                    | Astronomer Certified / Astro Runtime         | Helm |
| ------------------- | ---------------------- | ------------- | -------- | ----------------------------------------- | -------------------------------------------- | ---- |
| v0.25               | 1.17¹, 1.18¹, 1.19¹, 1.20¹, 1.21       | 0.25.x        | 9.6+     | 3.6, 3.7, 3.8, 3.9 (_requires AC 2.2.0+_) | All supported Certified versions*            | 3.6  |
| v0.28               | 1.19¹, 1.20¹, 1.21, 1.22, 1.23, 1.24 | 1.0.x - 1.1.x | 9.6+     | 3.6, 3.7, 3.8, 3.9 (_requires AC 2.2.0+_) | All supported Certified versions             | 3.6  |
| v0.29               | 1.19¹, 1.20¹, 1.21, 1.22, 1.23, 1.24 | 1.3.x         | 9.6+     | 3.6, 3.7, 3.8, 3.9 (_requires AC 2.2.0+_) | All supported Certified and Runtime versions | 3.6  |
| v0.30               | 1.19¹, 1.20¹, 1.21, 1.22, 1.23, 1.24 | 1.4.x - 1.7.x | 9.6+     | 3.6, 3.7, 3.8, 3.9 (_requires AC 2.2.0+_) | All supported Certified and Runtime versions | 3.6  |

:::info

> ¹ Some Kubernetes versions are compatible only up until a specific patch version of Astronomer Software.

Reviewer: ```suggestion
¹ Support for some Kubernetes versions is limited to a specific patch version of Astronomer Software.
```

> - Support for Kubernetes 1.17 ends with 0.25.12.

Reviewer: ```suggestion
- Support for Kubernetes 1.17 ends with Astronomer Software 0.25.12.
```

> - Support for Kubernetes 1.18 ends with 0.25.13.

Reviewer: ```suggestion
- Support for Kubernetes 1.18 ends with Astronomer Software 0.25.13.
```

- [Astro Runtime maintenance and lifecycle policy](runtime-version-lifecycle-policy.md)
- [Astronomer Certified versioning and support](ac-support-policy.md)

:::info

> Due to the [deprecation of Dockershim](https://kubernetes.io/blog/2020/12/02/dockershim-faq/), Azure does not support private CAs starting with Kubernetes 1.19. If you use a private CA, contact [Astronomer support](https://support.astronomer.io) before upgrading to Kubernetes 1.19 on AKS.

Reviewer: ```suggestion
Due to the [deprecation of Dockershim](https://kubernetes.io/blog/2020/12/02/dockershim-faq/), Azure does not support private Certificate Authorities (CAs) starting with Kubernetes 1.19. If your organization is using a private CA, contact [Astronomer support](https://support.astronomer.io) before upgrading to Kubernetes 1.19 on Azure Kubernetes Service (AKS).
```

:::

:::info

> While Astronomer v0.25 is compatible with Astronomer Certified 2.2.0, support for the Airflow triggerer is available only in Astronomer v0.26+. To use [deferrable operators](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html), which require the Airflow triggerer, you must upgrade.

Reviewer: ```suggestion
While Astronomer v0.25 is compatible with Astronomer Certified 2.2.0, support for the Airflow triggerer is available only in Astronomer v0.26 and later. To use [deferrable operators](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html), that require the Airflow triggerer, you must upgrade.
```
